cfgov/core/middleware.py

```diff
@@ -99,3 +100,29 @@ def __call__(self, request):
         response = self.get_response(request)
         translation.deactivate()
         return response
+
+
+class SelfHealingMiddleware:
```

Comment:
Let's copy over the docstring from the previous 404 handling method to retain the explanation of what exactly the intention is here.

```suggestion
class SelfHealingMiddleware:
    """Attempt to self-heal 404-ing URLs.	
    Takes a 404ing request and tries to transform it to a successful request	
    by lowercasing the path and stripping extraneous characters from the end.	
    If those result in a modified path, redirect to the modified path.	
    If the path did not change, this is a legitimate 404, so continue handling	
    that as normal.	
    """
```